src/app/server-cluster/AttributeListBuilder.h

```diff
@@ -38,21 +37,10 @@ class AttributeListBuilder
 public:
     AttributeListBuilder(ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder) : mBuilder(builder) {}
 
-    struct OptionalAttributeEntry
-    {
-        bool enabled;                               // Is this optional attribute enabled?
-        const DataModel::AttributeEntry & metadata; // Metadata for the attribute
-    };
-
-    /// Appends the given attributes to the builder.
-    ///
-    /// It is very common to have a set of mandatory and a set of optional attributes for a
-    /// cluster. This method allows for a single call to set up all of the given attributes in `builder`:
-    ///   - mandatoryAttributes
-    ///   - optionalAttributes IF AND ONLY IF they are enabled
-    ///   - all of `GlobalAttributes()`
+    /// appends the given attributes to the builder.
+    /// Optional attributes are only appended if the given `enabledOptionalAttributes` has them set
```

Comment:
The documentation here lost the bits about how this class will handle appending the global attributes.  That part should stay, no?